extern/dfelibs/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# CMake include(s).
+cmake_minimum_required( VERSION 3.11 )
+include( FetchContent )
+
+# Tell the user what's happening.
+message( STATUS "Building dfelibs as part of the TRACCC project" )
+
+# Declare where to get dfelibs from.
+set( TRACCC_DFELIBS_SOURCE
+   "URL;https://github.com/acts-project/dfelibs/archive/refs/tags/v20211029.tar.gz;URL_MD5;87fb09c5a11b98250f5e266e9cd501ea"
+   CACHE STRING "Source for dfelibs, when built as part of this project" )
+mark_as_advanced( TRACCC_DFELIBS_SOURCE )
+FetchContent_Declare( dfelibs ${TRACCC_DFELIBS_SOURCE} )
+
+# Options used in the build of dfelibs.
+set( dfelibs_BUILD_EXAMPLES FALSE CACHE BOOL
+   "Turn off the build of the dfelibs examples" )
+set( dfelibs_BUILD_UNITTESTS FALSE CACHE BOOL
+   "Turn off the build of the dfelibs unit tests" )
+set( dfelibs_ENABLE_INSTALL TRUE CACHE BOOL
+   "Have dfelibs be installed together with the rest of the project" )
```

Comment:
Is this necessary? I thought dfelibs was header only, why would we install it with the rest?

Reply:
The CSV reading code is all templated. And it uses dfelibs. Since we install the CSV reading headers, we must also install the dfelibs ones.

I had to do the same in Detray. The CSV reading code of that also depends on this external. If it didn't install dfelibs with itself (or found it from some other location), then `find_package(detray)` could not provide functional targets to the clients.

Reply:
I suppose this depends on how we plan to use traccc. I was always under the impression that it was a standalone demonstrator that nothing else would depend on, but I suppose it is more flexible if we install dfelibs...